<?xml version="1.0"?>
<robot name="jinwubot">

<material name="black">
 <color rgba="0.0 0.0 0.0 1.0"/>
</material>
<material name="orange">
 <color rgba="1.0 0.4 0.0 1.0"/>
</material>

  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.0" radius="0.0" />
      </geometry>
	<material name="black"/>
    </visual>
  </link>
  <link name="link1">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.11" radius="0.045" />
      </geometry>
	<material name="orange"/>
    </visual>
    <visual>
      <origin rpy="-1.5708 0 0" xyz="0 0.035 0"/>
      <geometry>
        <cylinder length="0.07" radius="0.045" />
      </geometry>
	<material name="orange"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.11" radius="0.045" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0 0.035 0"/>
      <geometry>
        <cylinder length="0.07" radius="0.045" />
      </geometry>
    </collision>
  </link>

  <link name="link2">
    <visual>
	<!-- center of cylinder in link frame -->
      <origin rpy="0 0 0" xyz="0 0 0.1275"/>
      <geometry>
        <cylinder length="0.115" radius="0.05" />
      </geometry>
	<material name="black"/>
    </visual>
    <visual>
	<!-- 0.56/2-0.048 -->
      <origin rpy="0 1.5708 0" xyz="0.287 0 0.1275"/>
      <geometry>
        <cylinder length="0.51" radius="0.045" />
      </geometry>
	<material name="black"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0.51 0 0.1375"/>
      <geometry>
        <cylinder length="0.095" radius="0.045" />
      </geometry>
	<material name="black"/>
    </visual>

    <collision>
	<!-- center of cylinder in link frame -->
      <origin rpy="0 0 0" xyz="0 0 0.1275"/>
      <geometry>
        <cylinder length="0.115" radius="0.05" />
      </geometry>
    </collision>
    <collision>
	<!-- 0.56/2-0.48+0.05 -->
      <origin rpy="0 1.5708 0" xyz="0.287 0 0.1275"/>
      <geometry>
        <cylinder length="0.56" radius="0.045" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.51 0 0.1375"/>
      <geometry>
        <cylinder length="0.095" radius="0.045" />
      </geometry>
    </collision>
  </link>
  <link name="link3" >
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <cylinder length="0.09" radius="0.045" />
      </geometry>
	<material name="orange"/>
    </visual>
    <visual>
      <origin rpy="0 1.5708 0" xyz="0.255 0 0.03"/>
      <geometry>
        <cylinder length="0.51" radius="0.035" />
      </geometry>
	<material name="orange"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0.51 0 0.04"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
	<material name="orange"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <cylinder length="0.09" radius="0.045" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 1.5708 0" xyz="0.255 0 0.03"/>
      <geometry>
        <cylinder length="0.51" radius="0.035" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.51 0 0.04"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
    </collision>
  </link>

  <link name="link4" >
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.035" />
      </geometry>
	<material name="black"/>
    </visual>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 0.0425 0"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
	<material name="black"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.035" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 0.0425 0"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
    </collision>
  </link>
  <link name="link5" >
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.035" />
      </geometry>
	<material name="orange"/>
    </visual>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0 -0.0425 0"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
	<material name="orange"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.035" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0 -0.0425 0"/>
      <geometry>
        <cylinder length="0.085" radius="0.035" />
      </geometry>
    </collision>
  </link>
  <link name="hand" >
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <box size="0.06 0.02 0.1" />
      </geometry>
	<material name="black"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="0.02 0 0.15"/>
      <geometry>
        <box size="0.01 0.02 0.1" />
      </geometry>
	<material name="black"/>
    </visual>
    <visual>
      <origin rpy="0 0 0" xyz="-0.02 0 0.15"/>
      <geometry>
        <box size="0.01 0.02 0.1" />
      </geometry>
	<material name="black"/>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <box size="0.06 0.02 0.1" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="0.02 0 0.15"/>
      <geometry>
        <box size="0.01 0.02 0.1" />
      </geometry>
    </collision>
    <collision>
      <origin rpy="0 0 0" xyz="-0.02 0 0.15"/>
      <geometry>
        <box size="0.01 0.02 0.1" />
      </geometry>
    </collision>
  </link>



  <joint name="joint0" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.048" rpy="0 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="104" velocity="0.3927" lower="-4.7124" upper="4.7124" />
  </joint>
  <joint name="joint1" type="revolute">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="0.0 0 0.0" rpy="-1.5708 -1.5708 0" />
    <axis xyz="0 0 1"/>
    <limit effort="104" velocity="0.3927" lower="-1.5708" upper="2.7925" />
  </joint>
  <joint name="joint2" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0.51 0 0.0" rpy=" 0 0 1.5708" />
    <axis xyz="0 0 1"/>
    <limit effort="104" velocity="0.6283" lower="-4.3633" upper="1.3090" />
  </joint>

  <joint name="joint3" type="revolute">
    <parent link="link3"/>
    <child link="link4"/>
    <origin xyz="0.51 0 0.11" rpy=" 0 0 -1.5708" />
    <axis xyz="0 0 1"/>
    <limit effort="20" velocity="0.7854" lower="-3.1416" upper="3.1416" />
  </joint>
  <joint name="joint4" type="revolute">
    <parent link="link4"/>
    <child link="link5"/>
    <origin xyz="0 0.08662 0" rpy="-1.5708 0 0" />
    <axis xyz="0 0 1"/>
    <limit effort="20" velocity="0.7854" lower="-3.1416" upper="3.1416" />
  </joint>
  <joint name="joint5" type="revolute">
    <parent link="link5"/>
    <child link="hand"/>
    <origin xyz="0 -0.035 0" rpy="1.5708 0 0 " />
    <axis xyz="0 0 1"/>
    <limit effort="20" velocity="0.7854" lower="-6.2832" upper="6.2832" />
  </joint>

</robot>
